pptx/text/bullets.py

```diff
@@ -0,0 +1,720 @@
+        return self._bullet.start_at
+    
+    @start_at.setter
+    def start_at(self, value):
+        """ Sets starting Value for AutoNumber Bullets """
+        self._bullet.start_at = value
+    
+    @property
+    def char(self):
+        """ String used as the bullet for a CharBullet"""
+        return self._bullet.character
+
+    @char.setter
+    def char(self, value):
+        """ String used as the bullet for a CharBullet"""
+        self._bullet.char = value
+
+
+    @property
+    def type(self):
+        """ Return a string type """
+        return self._bullet.type
+    
+
+class _Bullet(object):
+    """
+    Object factory for TextBullet objects
+    """
+
+    def __new__(cls, xBullet):
+        if isinstance(xBullet, CT_TextNoBullet):
+            bullet_cls = _NoBullet
+        elif isinstance(xBullet, CT_TextBulletAutoNumber):
+            bullet_cls = _AutoNumBullet
+        elif isinstance(xBullet, CT_TextCharBullet):
+            bullet_cls = _CharBullet
+        elif isinstance(xBullet, CT_TextBlipBullet):
+            bullet_cls = _PictureBullet
+        else:
+            bullet_cls = _NoBullet
+        
+        return super(_Bullet, cls).__new__(bullet_cls)
+
+    @property
+    def char(self):
+        """ Raise TypeError for types that do not override this property"""
+        tmpl = "TextBullet type %s has no char property. call .character() first"
+        raise TypeError(tmpl % self.__class__.__name__)
+
+    @char.setter
+    def char(self, value):
+        """ Raise TypeError for types that do not override this property"""
+        tmpl = "TextBullet type %s has no char property. call .character() first"
+        raise TypeError(tmpl % self.__class__.__name__)
+
+    @property
+    def char_type(self):
+        """ Raise TypeError for types that do not override this property"""
+        tmpl = "TextBullet type %s has no char_type property. call .auto_number() first"
+        raise TypeError(tmpl % self.__class__.__name__)
+
+    @char_type.setter
+    def char_type(self, value):
+        """ Raise TypeError for types that do not override this property"""
+        tmpl = "TextBullet type %s has no char_type property. call .auto_number() first"
+        raise TypeError(tmpl % self.__class__.__name__)
+
+    @property
+    def start_at(self):
+        """ Raise TypeError for types that do not override this property"""
+        tmpl = "TextBullet type %s has no start_at property. call .auto_number() first"
+        raise TypeError(tmpl % self.__class__.__name__)
+
+    @start_at.setter
+    def start_at(self, value):
+        """ Raise TypeError for types that do not override this property"""
+        tmpl = "TextBullet type %s has no start_at property. call .auto_number() first"
+        raise TypeError(tmpl % self.__class__.__name__)
+
+    @property
+    def type(self):
+        tmpl = ".type property must be implmented on %s"
+        raise NotImplementedError(tmpl % self.__class__.__name__)
```

Comment:
So in most cases throughout the library this actually returns a value of an enum rather than a string.  It also uses the `type` property and the enum value returned in the Factory Class to determine which child class to use.  Since we don't have separate enum's for each of these (I looked and couldn't find a schema for any of them), I just decided to use text strings. Honestly, none of these `type` properties are ever used at the moment so I actually considered removing them entirely but decided to leave them in.

Reply:
The only place I use .type for any object in python-pptx is for testing. I don't think we ever use them in the production code. It might be good to keep them for testing purposes.